pkg/il/read.go

```diff
@@ -0,0 +1,375 @@
+// Copyright 2017 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package il
```

Comment:
On second thought this would cause a bit of pain. For one, the code in conversion.go is shared. Those need to be made public (can be done, but it is really implementation detail). Second, the builder tests are based on reader/writers and for some reason referencing il/text from il/ tests are causing compiler issues:

/il/small_tests.a.GoCompileFile.params' failed: Process exited with status 1 [sandboxed].
istio.io/mixer/pkg/il/builder_test.go:350: cannot use p (type *Program) as type *il.Program in argument to text.WriteText
Use --strategy=GoCompile=standalone to disable sandboxing for the failing actions.